@re-/model/demos/Stack.ts

```diff
@@ -0,0 +1,66 @@
+interface IStack<T> {
+    push(item: T): void
+    pop(): T | undefined
+    peek(): T | undefined
+    size(): number
+}
+
+const DEFAULT_STACK_CAPACITY = 30
+const DELIMITER = "$$block_start$$"
+
+export class Stack<T> implements IStack<string> {
```

Comment:
Let's call this something more specific like `StatementStack` since it contains logic specific to dealing with statements.